src/SkeletalTrapezoidation.h

```diff
@@ -0,0 +1,427 @@
+     */
+    bool filterUnmarkedRegions(edge_t* to_edge, coord_t bead_count, coord_t traveled_dist, coord_t max_dist, const BeadingStrategy& beading_strategy);
+
+    /*!
+     * Representing the location along an edge where the anchor position of a transition should be placed.
+     */
+    struct TransitionMiddle
+    {
+        coord_t pos; //! position along edge as measure from edge.from.p
+        coord_t lower_bead_count;
+        TransitionMiddle(coord_t pos, coord_t lower_bead_count)
+        : pos(pos), lower_bead_count(lower_bead_count)
+        {}
+    };
+
+    /*!
+     * Auxiliary for referencing one transition along an edge which may contain multiple transitions
+     */
+    struct TransitionMidRef
+    {
+        std::unordered_map<edge_t*, std::list<TransitionMiddle>>::iterator pair_it;
+        std::list<TransitionMiddle>::iterator transition_it;
+        TransitionMidRef(std::unordered_map<edge_t*, std::list<TransitionMiddle>>::iterator pair_it, std::list<TransitionMiddle>::iterator transition_it)
+        : pair_it(pair_it)
+        , transition_it(transition_it)
+        {}
+    };
+
+    /*!
+     * Represents the location along an edge where the lower or upper end of a transition should be placed.
+     */
+    struct TransitionEnd
+    {
+        coord_t pos; //!< position along edge as measure from edge.from.p, where the edge is always the half edge oriented from lower to higher R
+        coord_t lower_bead_count;
+        bool is_lower_end; //!< whether this is the ed of the transition with lower bead count
+        TransitionEnd(coord_t pos, coord_t lower_bead_count, bool is_lower_end)
+        : pos(pos), lower_bead_count(lower_bead_count), is_lower_end(is_lower_end)
+        {}
+    };
+
+    void generateTransitionMids(const BeadingStrategy& beading_strategy, std::unordered_map<edge_t*, std::list<TransitionMiddle>>& edge_to_transitions);
+
+    void filterTransitionMids(std::unordered_map<edge_t*, std::list<TransitionMiddle>>& edge_to_transitions, const BeadingStrategy& beading_strategy);
+
+    /*!
+     * 
+     * \param edge_to_start edge pointing to the node from which to start traveling in all directions except along \p edge_to_start
+     * \param origin_transition The transition for which we are checking nearby transitions
+     * \param traveled_dist the distance traveled before we came to \p edge_to_start.to
+     * \param going_up Whether we are traveling in the upward direction as seen from the \p origin_transition. If this doesn't align with the direction according to the R diff on a consecutive edge we know there was a local optimum
+     * \return whether the origin transition should be dissolved
+     */
+    std::list<TransitionMidRef> dissolveNearbyTransitions(edge_t* edge_to_start, TransitionMiddle& origin_transition, coord_t traveled_dist, coord_t max_dist, bool going_up, std::unordered_map<edge_t*, std::list<TransitionMiddle>>& edge_to_transitions, const BeadingStrategy& beading_strategy);
```

Comment:
Like I said below, I don;t think that's a good idea, since `splice` is used. Lists are great for that because once you have the iterator, they have a constant insert time.